examples/summarization/t5/evaluate_cnn.py

```diff
@@ -0,0 +1,83 @@
+import argparse
+from pathlib import Path
+
+from rouge_score import rouge_scorer, scoring
+from tqdm import tqdm
+
+from transformers import T5Tokenizer, TFT5ForConditionalGeneration
+
+
+def chunks(lst, n):
+    """Yield successive n-sized chunks from lst."""
+    for i in range(0, len(lst), n):
+        yield lst[i : i + n]
+
+
+def generate_summaries(lns, out_file, batch_size):
+    fout = Path(out_file).open("w")
+
+    model = TFT5ForConditionalGeneration.from_pretrained("t5-large")
+    tokenizer = T5Tokenizer.from_pretrained("t5-large")
+
+    # update config with summarization specific params
+    task_specific_params = model.config.task_specific_params
+    if task_specific_params is not None:
+        model.config.update(task_specific_params.get("summarization", {}))
+
+    for batch in tqdm(list(chunks(lns, batch_size))):
+        batch = [model.config.prefix + text for text in batch]
+
+        dct = tokenizer.batch_encode_plus(batch, max_length=512, return_tensors="tf", pad_to_max_length=True)
+        summaries = model.generate(input_ids=dct["input_ids"], attention_mask=dct["attention_mask"],)
+        dec = [tokenizer.decode(g, skip_special_tokens=True, clean_up_tokenization_spaces=False) for g in summaries]
+
+        for hypothesis in dec:
+            fout.write(hypothesis + "\n")
+            fout.flush()
+
+
+def calculate_rouge(output_lns, reference_lns, score_path):
+    score_file = Path(score_path).open("w")
+    scorer = rouge_scorer.RougeScorer(["rouge1", "rouge2", "rougeL"], use_stemmer=True)
+    aggregator = scoring.BootstrapAggregator()
+
+    for reference_ln, output_ln in zip(reference_lns, output_lns):
+        scores = scorer.score(reference_ln, output_ln)
+        aggregator.add_scores(scores)
+
+    result = aggregator.aggregate()
+    score_file.write(
+        "ROUGE_1: \n{} \n\n ROUGE_2: \n{} \n\n ROUGE_L: \n{} \n\n".format(
+            result["rouge1"], result["rouge2"], result["rougeL"]
+        )
+    )
+
+
+def _run_generate():
```

Comment:
a) probably shouldn't be private (feel free to fix mine also
b) Would be down to share a function `get_summarization_args` to guarantee that we have the same command line interface moving forward.